### PR TITLE
Fix `EditorInterface.get_selected_paths()` working incorrectly when FileSystemDock is in split mode

### LINKED ISSUE #88228: `EditorInterface.get_selected_paths()` works incorrectly when FileSystemDock is in split mode
### Tested versions

- Reproduced in: 4.2.1.stable

### System information

Windows 10 - Godot 4.2.1.stable

### Issue description

The documentation for `EditorInterface.get_selected_paths()` states that it ["Returns an array containing the paths of the currently selected files (and directories) in the FileSystemDock](https://docs.godotengine.org/en/stable/classes/class_editorinterface.html#class-editorinterface-method-get-selected-paths). However, when the FileSystemDock is in split mode (i.e. the folder structure displaying separately from the files), this only shows the selected folder in the folder structure display.
Example:
image
When runner.gd is selected in the FileSystemDock and `EditorInterface.get_selected_paths()` is called, this only returns: `["res://"]`. The MRP runs this through a simple EditorScript:
```
@tool
extends EditorScript

func _run():
	print(EditorInterface.get_selected_paths())
```
Even when selecting multiple files within `res://`, this returns the exact same result: `["res://"]`.
On the other hand: `EditorInterface.get_current_path()` returns the correct currently selected file, even in split mode. This, however, does not support selecting multiple files.

### Steps to reproduce

1. Open MRP
2. Enter split mode through the following button on the FileSystemDock:
image
3. Select any files in `res://` in the FileSystemDock
4. run `runner.gd` in the script editor tab with `CTRL+Shift+X`

### Minimal reproduction project (MRP)

[MRP.zip](https://github.com/godotengine/godot/files/14234535/MRP.zip)

### PR DESCRIPTION
**Pull request by fstxz**
Fixes #88228

### CODE REVIEW

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -2049,6 +2056,15 @@ Vector<String> FileSystemDock::_tree_get_selected(bool remove_self_inclusion, bo
 	return selected_strings;
 }
 
+Vector<String> FileSystemDock::_file_list_get_selected() const {
+	Vector<String> selected;
+
+	for (const int idx : files->get_selected_items()) {
```
Comment (AThousandShips) [inline editor/filesystem_dock.cpp:2062]:
```suggestion
	for (int idx : files->get_selected_items()) {
```
We don't normally add `const` to simple types like this

**Comment by AThousandShips** [comment]
Will try and test this tomorrow

File: editor/filesystem_dock.cpp [hunk 2]
```diff
@@ -685,7 +685,13 @@ void FileSystemDock::_tree_multi_selected(Object *p_item, int p_column, bool p_s
...
 
 Vector<String> FileSystemDock::get_selected_paths() const {
-	return _tree_get_selected(false);
+	Vector<String> selected;
+	if (display_mode == DISPLAY_MODE_TREE_ONLY) {
+		selected = _tree_get_selected(false);
+	} else {
+		selected = _file_list_get_selected();
+	}
+	return selected;
```
Comment (AThousandShips) [inline editor/filesystem_dock.cpp:694]:
```suggestion
	if (display_mode == DISPLAY_MODE_TREE_ONLY) {
		return _tree_get_selected(false);
	} else {
		return _file_list_get_selected();
	}
```

Comment (fstxz) [reply editor/filesystem_dock.cpp]: Oh boy I'm so dumb

**Comment by fstxz** [comment]
I wonder if folders that are selected in the tree should also be included when in split mode? Otherwise you can't select multiple folders that are in different places without changing the display mode.

**Review by AThousandShips — APPROVED** [review]
I've confirmed the bug and confirmed that this PR works correctly in both modes (i.e. doesn't regress with the single tree mode selected)

I'd say the expected behavior is that the list of files is returned, but the option of including the folder as well might be considered, though I think we can go with this and see what we might want later as well (we could add a note to the documentation clarifying what exactly is returned, elaborating that it returns just the files selected in split mode)

Approving on that this works and the code looks good, might be a more elegant solution but I'm not familiar with the code here so can't speak to that

**Comment by KoBeWi** [comment]
In split mode if you have a folder selected, but none of the files, the returned list is empty. If no file is selected, maybe the selected directory should be returned at least?

**Comment by fstxz** [comment]
@KoBeWi I updated the PR, I hope this is what you meant.

**Review by KoBeWi — APPROVED** [review]
Seems like it's possible to multi-select in tree in split view, so maybe all selected directories should be returned 🤔
But this way it's fine too.

**Comment by fstxz** [comment]
> Seems like it's possible to multi-select in tree in split view

It's also possible to deselect everything with ctrl, this is why I chose to return the current directory.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by akien-mga** [comment]
Cherry-picked for 4.3.1.

**Pull request merged**